src/test/java/codeu/controller/ActivityFeedServletTest.java

```diff
@@ -97,30 +100,54 @@ public void testDoGet() throws IOException, ServletException {
           .thenReturn(fakeAllConversations);
 
       
-      List<List<Message>> fakeAllMessages = new ArrayList<>();
-
-      List<Message> fakeMessageList = new ArrayList<>();
-      fakeMessageList.add(
+      List<Message> fakeAllMessages = new ArrayList<>();
+      fakeAllMessages.add(
           new Message(
               UUID.randomUUID(),
               fakeConversationId,
               fakeUserId,
               "test message",
               Instant.now()));
+      Mockito.when(mockMessageStore.getMessagesInConversation(fakeConversationId))
+          .thenReturn(fakeAllMessages);
       
-      fakeAllMessages.add(fakeMessageList);
+      Hashtable<Message, String> fakeMessageToConversationTitle = new Hashtable();
```

Comment:
This test is now duplicating the same logic that you have in the main action. It's usually better for tests to pick a few hardcoded examples and make sure that you get the output you would expect. Otherwise, if there's a problem with your getCreation time method in the main class, there might be the same problem in this test!

So for this case, one thing you want to test is that objects with different time stamps are set in the right order. You can use `Instant.fromEpochSeconds()` to set the creation timestamps on your fake objects, then use a Mockito `ArgumentCaptor` to look at the request attributes and make sure that the order matches what you expect.

I'm always happy to talk more about test style and best practices, and I know Mockito can be a tricky tool to learn, so let me know if you want to set up some time to discuss further. :+1: